### PR TITLE
7238 absolute number correspondence conversion

### PR DESCRIPTION
**Pull request by otavioabreu27**
### Describe your change: Creation of the absolute_conversion function
Fixes: #7238 

The function takes 3 arguments:
- interval_start: The start of the interval 
- interval_end: The end of the interval
- number: Number to be compared 

The program converts the given interval into a absolute 0 to a 100 one and returns the correspondent value in percentage.
It's a usefull function to problems that take scales with shifting intervals and need to make comparisons between them.

* [X] Add an algorithm?
* [  ] Fix a bug or typo in an existing algorithm?
* [  ] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: conversions/absolute_conversion.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    >>> absolute_conversion(0, 10, 4)
+    '40.0'
+    >>> absolute_conversion(120, 140, 125)
+    '25.0'
+    >>> absolute_conversion(0,20,21)
+    '105.0'
+    """
+    if interv_start > interv_end:
+        message = "Invalid arguments, start is higher than the end."
+        return message
```
Comment (CaedenPH) [inline conversions/absolute_conversion.py:37]: You should raise an error instead of returning a message

File: conversions/absolute_conversion.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+
+def absolute_conversion(interv_start: float, interv_end: float, number: float) -> str:
+    """
+    >>> absolute_conversion(0, 10, 4)
+    '40.0'
+    >>> absolute_conversion(120, 140, 125)
+    '25.0'
+    >>> absolute_conversion(0,20,21)
+    '105.0'
```
Comment (CaedenPH) [inline conversions/absolute_conversion.py:33]: You should add a doctest that returns an error

File: conversions/absolute_conversion.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+"""
+
+
+def absolute_conversion(interv_start: float, interv_end: float, number: float) -> str:
+    """
+    >>> absolute_conversion(0, 10, 4)
+    '40.0'
+    >>> absolute_conversion(120, 140, 125)
+    '25.0'
+    >>> absolute_conversion(0,20,21)
```
Comment (CaedenPH) [inline conversions/absolute_conversion.py:32]:
```suggestion
    >>> absolute_conversion(0, 20, 21)
```

File: conversions/absolute_conversion.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    interv_end:100
+    number:50
+    output: 50.0
+    interv_start:6
+    interv_end:12
+    number:9
+    output:50.0
+
+    Link: https://en.wikipedia.org/wiki/Conversion_of_units
+"""
```
Comment (CaedenPH) [inline conversions/absolute_conversion.py:23]:
```suggestion
"""
Convert any interval into a 0 to 100 scale and see a respective number in between
percentage of the whole.
Installation:
    - Import this file
    - Call the absolute_conversion function
    - Parameters:
        - interv_start: The start of your internal scale.
        - interv_end: The end of your internal scale.
        - number: The number you want to know the percentage that it
        represents of the scale.
Examples:
interv_start:0
interv_end:100
number:50
output: 50.0
interv_start:6
interv_end:12
number:9

Link: https://en.wikipedia.org/wiki/Conversion_of_units
"""
```

File: conversions/absolute_conversion.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    if interv_start > interv_end:
+        message = "Invalid arguments, start is higher than the end."
+        return message
+    elif interv_start == interv_end:
+        message = "Invalid arguments, start and end should be different."
+        return message
+    else:
+        native_interval = interv_end - interv_start
+        value_percentage = (100 * (number - interv_start)) / native_interval
+        return str(value_percentage)
```
Comment (CaedenPH) [inline conversions/absolute_conversion.py:44]: Why are you casting the percent to string?

**Comment by tianyizheng02** [comment]
Hi @otavioabreu27, thanks for your contribution, but I just noticed that the algorithm you propose is nearly identical to the `normalization` function we already have in `machine_learning/data_transformations.py`:

```py
native_interval = interv_end - interv_start
value_percentage = (100 * (number - interv_start)) / native_interval
return str(value_percentage)
```

https://github.com/TheAlgorithms/Python/blob/ce218c57f1f494cfca69bc01ba660c97385e5330/machine_learning/data_transformations.py#L44-L47

The main differences between your contribution and `normalization` are
1) `normalization` takes a list of numbers as input rather than a single number,
2) `normalization` assumes the interval bounds to be the min and max of the list, and
3) your contribution scales the computed value up by a factor of 100.

However, I hope you can see that the normalization calculation itself is fundamentally identical to yours. We'd like to avoid duplicate algorithms in the repo, so I'd appreciate it if you could make your contribution to `machine_learning/data_transformations.py` instead. Your contributions here could easily enhance `normalization`'s implementation by allowing the user to set custom interval bounds.

**Pull request closed without merging**